### PR TITLE
Expose video view parameters android

### PR DESCRIPTION
**Pull request by mpozy**

### CODE REVIEW

File: lib/src/sdk_api/view/video_view.dart [hunk 1]
```diff
@@ -214,6 +214,7 @@ class _VideoViewState extends State<VideoView> {
                     MethodChannel("video_view_${id}_method_channel");
                 _methodChannel = channel;
                 channel.invokeMethod("attach", creationParams);
+                channel.invokeMethod("setScaleType", videoViewParams);
```
Comment (graduad) [inline lib/src/sdk_api/view/video_view.dart:217]: Calling 2 methods here is not ideal.

File: lib/src/sdk_api/view/video_view.dart [hunk 2]
```diff
@@ -161,6 +159,7 @@ class _VideoViewState extends State<VideoView> {
   @override
   Widget build(BuildContext context) {
     final Map<String, String> creationParams = {};
+    final Map<String, String> videoViewParams = {};
```
Comment (graduad) [inline lib/src/sdk_api/view/video_view.dart:162]: `creationParams` was supposed to contain all the parameters required for the creation of the view. Perhaps if we call only one method to initialise the native view we can achieve this. Please see the comments stating "Calling 2 methods here is not ideal."

**Review by graduad — CHANGES_REQUESTED** [review]

File: android/src/main/kotlin/io/dolby/comms/sdk/flutter/NativeView.kt [hunk 3]
```diff
@@ -71,12 +71,10 @@ class NativeView(context: Context, id: Int, messenger: BinaryMessenger): Platfor
     }
 
     fun setScaleType(scaleType: String?) {
```
Comment (graduad) [inline android/src/main/kotlin/io/dolby/comms/sdk/flutter/NativeView.kt:73]: Does `scaletype: String?` need to be an optional?

File: lib/src/sdk_api/view/video_view.dart [hunk 4]
```diff
@@ -105,18 +101,20 @@ class VideoView extends StatefulWidget {
   const VideoView.withMediaStream(
       {required this.participant,
       required this.mediaStream,
-      this.scaleType,
```
Comment (graduad) [inline lib/src/sdk_api/view/video_view.dart:108]: Does this have to be an optional? Can a default value not be provided?

Would following not work?

```dart
const VideoView.withMediaStream({
       required this.participant,
       required this.mediaStream,
       ScaleType scaleType = ScaleType.fill,
       Key? key})
```

File: lib/src/sdk_api/view/video_view.dart [hunk 5]
```diff
@@ -105,18 +101,20 @@ class VideoView extends StatefulWidget {
...
 
   /// A constructor that shuold be used when the [VideoView] is used as a stand-alone widget
   /// outside of collection widgets such as [GridView] or [ListView]. The constructor requires
   /// providing the [VideoViewController] and, optionally, a [Key].
-  const VideoView({required this.videoViewController, Key? key})
-      : participant = null,
+  const VideoView(
+      {required this.videoViewController, ScaleType? scaleType, Key? key})
+      : scaleType = scaleType ?? ScaleType.fill,
+        participant = null,
```
Comment (graduad) [inline lib/src/sdk_api/view/video_view.dart:116]: Same as above.

**Review by graduad — CHANGES_REQUESTED** [review]

File: lib/src/sdk_api/view/video_view.dart (hunk 4 above)
Comment (mpozy) [reply lib/src/sdk_api/view/video_view.dart:108]: Default value is set when value is not provided. In above example scaleType will be always ScaleType.fill.

File: lib/src/sdk_api/view/video_view.dart (hunk 5 above)
Comment (mpozy) [reply lib/src/sdk_api/view/video_view.dart:116]: Same as above.

File: lib/src/sdk_api/view/video_view.dart (hunk 4 above)
Comment (mpozy) [reply lib/src/sdk_api/view/video_view.dart:108]: ScaleType could be required, but then you have to provide scaleType and default value is not needed.

File: android/src/main/kotlin/io/dolby/comms/sdk/flutter/NativeView.kt (hunk 3 above)
Comment (mpozy) [reply android/src/main/kotlin/io/dolby/comms/sdk/flutter/NativeView.kt:73]: As far I can see all of creationParameters read from flutter are nullable.

Comment (graduad) [reply android/src/main/kotlin/io/dolby/comms/sdk/flutter/NativeView.kt:73]: Ok. I will resolve the conversation.

**Pull request merged**